Modules/socketmodule.c

```diff
@@ -5458,6 +5458,31 @@ PyDoc_STRVAR(getprotobyname_doc,
 \n\
 Return the protocol number for the named protocol.  (Rarely used.)");
 
+static PyObject *
+socket_close(PyObject *self, PyObject *fdobj)
+{
+    SOCKET_T fd;
+    int res;
+
+    fd = PyLong_AsSocket_t(fdobj);
+    if (fd == (SOCKET_T)(-1) && PyErr_Occurred())
+        return NULL;
+    Py_BEGIN_ALLOW_THREADS
+    res = SOCKETCLOSE(fd);
+    Py_END_ALLOW_THREADS
+    /* bpo-30319: The peer can already have closed the connection.
+       Python ignores ECONNRESET on close(). */
+    if (res < 0 && errno != ECONNRESET) {
```

Comment:
I think this should be `!CHECK_ERRNO(ECONNRESET)`, or use `GET_SOCK_ERROR` instead of `errno`. 

Winsock is an OS library, so it can't use `errno` of a particular C runtime library. It uses Windows API `[WSA]GetLastError` and `[WSA]SetLastError`.

Reply:
Good point!